src/compiler/emitter.ts

```diff
@@ -4406,15 +4406,15 @@ namespace ts {
             }
         }
 
-        function emitList(parentNode: Node | undefined, children: NodeArray<Node> | undefined, format: ListFormat, parenthesizerRule?: ParenthesizerRuleOrSelector<Node>, start?: number, count?: number) {
+        function emitList<Parent extends Node, Child extends Node, Children extends NodeArray<Child>>(parentNode: Parent | undefined, children: Children | undefined, format: ListFormat, parenthesizerRule?: ParenthesizerRuleOrSelector<Child>, start?: number, count?: number) {

```

Comment:
Why is `Parent` parameterized? It seems to be referenced only once.

Reply:
Has to be leftover cruft, oops. I'll remove this and the other cases.